packages/governance/src/types.js

```diff
@@ -1,77 +1,138 @@
- *
- * @typedef {{ [updater: string]: (arg: ParamValue) => void }} ParamManagerUpdaters
- * @typedef {ParamManagerBase & ParamManagerUpdaters} ParamManagerFull
+ * @typedef {Object} SimpleQuestion
+ * @property {string} text
  */
 
 /**
- * @typedef {Array<ParamDescription>} ParamDescriptions
+ * @typedef { Amount | Brand | Installation | Instance | bigint |
+ *   Ratio | string | unknown } ParamValue
  */
 
 /**
- * @callback BuildParamManager
- * @param {ParamDescriptions} paramDesc
- * @returns {ParamManagerFull}
+ * @template T
+ * @typedef {{ type: T, name: string }} ParamRecord
  */
 
 /**
- * @typedef {Object} QuestionTermsShort
- *   BallotDetails as provided to the Registrar
- * @property {ChoiceMethod} method
- * @property {string} question
- * @property {string[]} positions
- * @property {number} maxChoices
- * @property {ClosingRule} closingRule
+ * @typedef {ParamRecord<'amount'> & { value: Amount } |
+ *   ParamRecord<'brand'> & { value: Brand } |
+ *   ParamRecord<'installation'> & { value: Installation } |
+ *   ParamRecord<'instance'> & { value: Instance } |
+ *   ParamRecord<'nat'> & { value: bigint } |
+ *   ParamRecord<'ratio'> & { value: Ratio } |
+ *   ParamRecord<'string'> & { value: string } |
+ *   ParamRecord<'unknown'> & { value: unknown }
+ * } ParamDescription
  */
 
 /**
- * @typedef {Object} QuestionTerms
- *   BallotDetails after the Registrar adds its Instance
- * @property {ChoiceMethod} method
- * @property {string} question
- * @property {string[]} positions
- * @property {number} maxChoices
- * @property {ClosingRule} closingRule
+ * @typedef { SimpleQuestion | ParamChangeQuestion } Question
+ */
+
+/**
+ * @typedef {Object} QuestionTerms - BallotSpec plus the Registrar Instance and
+ *   a numerical threshold for the quorum. (The ballotCounter doesn't know the
+ *   size of the electorate, so the Registrar has to say what limit to enforce.)
+ * @property {BallotSpec} ballotSpec
+ * @property {number} quorumThreshold
  * @property {Instance} registrar
  */
+
 /**
- * @typedef {Object} BallotDetails
- *   BallotDetails after the Registrar adds its Instance
+ * @typedef {Object} TextPosition
+ * @property {string} text
+ */
+
+/**
+ * @typedef { TextPosition | ChangeParamPosition |
+ *   NoChangeParamPosition } Position
+ */
+
+/**
+ * @typedef {Object} BallotSpec
+ *   Specification when requesting a Ballot
  * @property {ChoiceMethod} method
- * @property {string} question
- * @property {string[]} positions
+ * @property {Question} question
+ * @property {Position[]} positions
+ * @property {ElectionType} electionType
  * @property {number} maxChoices
+ * @property {ClosingRule} closingRule
+ * @property {QuorumRule} quorumRule
+ * @property {Position} tieOutcome
+ */
+
+/**
+ * @typedef {BallotSpec} BallotDetails
+ *    complete ballot details: ballotSpec plus counter and handle
+ * @property {Instance} counterInstance - instance of the BallotCounter
+ * @property {Handle<'BallotTemplate'>} handle
  */
```

Comment:
From @michaelfig 's advice at https://github.com/Agoric/agoric-sdk/pull/3448#discussion_r700542501
I tried it and it seems to work.

```suggestion
/**
 * @typedef {Object} BallotDetailsExtraProperties
 * @property {Instance} counterInstance - instance of the BallotCounter
 * @property {Handle<'BallotTemplate'>} handle
 */

/**
 * @typedef {BallotSpec & BallotDetailsExtraProperties} BallotDetails
 *    complete ballot details: ballotSpec plus counter and handle
 */

```

Reply:
Works for tsc and VSCode, but WebStorm doesn't get it.  `grumble`  oh, well.